pkg/ibmcsidriver/controller.go

```diff
@@ -437,6 +438,13 @@ func (csiCS *CSIControllerServer) CreateSnapshot(ctx context.Context, req *csi.C
 	ctxLogger.Info("CSIControllerServer-CreateSnapshot... ", zap.Reflect("Request", *req))
 	defer metrics.UpdateDurationFromStart(ctxLogger, "CreateSnapshot", time.Now())
 
+	//Feature flag to enable/disable CreateSnapshot feature.
+	if strings.ToLower(os.Getenv("IS_SNAPSHOT_ENABLED")) == "false" {
+		ctxLogger.Warn("CreateSnapshot functionality is disabled.")
```

Comment:
Please clarify one thing here, does kubernetes wait until we return response from driver and then retry?

Reply:
Or does it also have some time context set, like it waits for only certain time like 5 mins and if doesn't receive any response, does it retry?

Reply:
Kubernetes waits for max 15 mins which is set as --timeout flag or any less time what driver is responding. So if driver waits for any time less than 15 mins in our case k8s wont retry else it will give error in csi-snapshotter as context deadline exceeded and then retry. We have bug in k8s csi-snapshooter they are not honouring exponential retry.

We are just delaying the back to back calls from k8s.